tests/unit/mathematics/linalg/ProbingSampler_unittest.cc

```diff
@@ -106,8 +106,9 @@ TEST(ProbingSampler, probing_samples_big_diag_matrix)
 
 	// test coloring stuffs
 	SGVector<int32_t> coloring_vector=trace_sampler->get_coloring_vector();
+
 	EXPECT_EQ(trace_sampler->get_num_samples(), 1);
-	for (index_t i=0; i<coloring_vector.vlen; ++i)
+	for (int32_t i = 0; i < coloring_vector.vlen; ++i)
```

Comment:
the `i` here should be the type of `index_t` as you are comparing it with `coloring_vector.vlen` which is `index_t`